### ISSUE TITLE
Internal error: unreachable code in `codegen/safepoint-table.cc` 

### ISSUE THREAD

**Issue by penguinwu** [v8-riscv/v8#56]
**Describe the bug**
Running `mjsunit/es6/block-conflicts-sloppy.js` triggers an internal error of `unreachable code`

**To Reproduce**
1. Build with simulated and debug build (see instruction [here](https://github.com/v8-riscv/v8/wiki/simulator-build))
2. Run test/program
```
out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/mjsunit/es6/block-conflicts-sloppy.js --random-seed=1881563104 --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner --validate-asm --allow-natives-syntax
```
```
#
# Fatal error in ../../src/codegen/safepoint-table.cc, line 62
# unreachable code
#
#
#
#FailureMessage Object: 0x7ffefb8d51d0
```

**Expected behavior**
No assert errors or other error messages if test passes.

**Additional context**

Another test fails in a similar way.

To reproduce:
```
out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/mjsunit/wasm/shared-memory-gc-stress.js --random-seed=-1611652631 --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner
```

Output:
```
iteration 6125
iteration 6150
iteration 6175
#
# Fatal error in ../../src/codegen/safepoint-table.cc, line 62
# unreachable code
#
#
#
#FailureMessage Object: 0x7fff56b5f230
```

**Comment by fw-brice** [comment]
This same error occurs for `es6/promises`:

```
out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/mjsunit/es6/promises.js --randomseed=1058420632 --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner --allow-natives-syntax
```

**Comment by fw-brice** [comment]
If I revert the changes recommended by @luyahan in #79, then `es6/promises` passes for me too, but `es6/block-conflicts-sloppy` still fails with the same error.

**Comment by derekzTu** [comment]
This same error appears for `test/debugger/debug/side-effect/debug-evaluate-no-side-effect-builtins.js`
```bash
out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/debugger/test-api.js test/debugger/debug/side-effect/debug-evaluate-no-side-effect-builtins.js --random-seed=-1971205773 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort
--enable-slow-asserts --verify-heap --testing-d8-test-runner --no-enable-one-shot-optimization --enable-inspector --allow-natives-syntax
```

and

`test/debugger/debug/debug-negative-break-points.js`
```bash
out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/debugger/test-api.js test/debugger/debug/debug-negative-break-points.js --random-seed=-1971205773 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --enable-slow-asserts -
-verify-heap --testing-d8-test-runner --enable-inspector --allow-natives-syntax
```

and

`test/debugger/debug/debug-evaluate.js`

```bash
out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/debugger/test-api.js test/debugger/debug/debug-evaluate.js --random-seed=-1971205773 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --enable-slow-asserts --verify-heap
--testing-d8-test-runner --enable-inspector --allow-natives-syntax
```

and

`test/debugger/debug/debug-scopes.js`
```bash
out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/debugger/test-api.js test/debugger/debug/debug-scopes.js --random-seed=-1971205773 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --enable-slow-asserts --verify-heap --
testing-d8-test-runner --noanalyze-environment-liveness --enable-inspector --allow-natives-syntax
```

**Comment by penguinwu** [comment]
@fw-brice Since several failures are traced back to this internal assert, could you coordinate w/ @DrRezaYazdani1  and @luyahan to give this issue a higher priority?

**Comment by derekzTu** [comment]
The error also pops up for debug + stress options for these tests:

Test: 
`test/mjsunit/eval-stack-trace.js`
Command:
```bash
out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/mjsunit/eval-stack-trace.js --random-seed=-827882926 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner
```
-----------
Test: 
`test/mjsunit/mjsunit-assertion-error.js`
Command:
```bash
out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/mjsunit/mjsunit-assertion-error.js --random-seed=-827882926 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner
```
-----------
Test:
`test/mjsunit/modules-init3.mjs`
Command:
```bash
out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/mjsunit/modules-init3.mjs --random-seed=-827882926 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner
```
-----------
Test:
`test/mjsunit/modules-namespace4.mjs`
Command:
```bash
out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/mjsunit/modules-namespace4.mjs --random-seed=-827882926 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner
```
-----------

Test:
`test/mjsunit/simple-constructor.js`
Command:
```bash
out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/mjsunit/simple-constructor.js --random-seed=-827882926 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner
```
-----------

Test:
`test/mjsunit/es6/block-let-declaration.js`
Command:
```bash
out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/mjsunit/es6/block-let-declaration.js --random-seed=-827882926 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner
```
-----------

Test:
`test/mjsunit/es6/classes-constructor.js`
Command:
```bash
out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/mjsunit/es6/classes-constructor.js --random-seed=-827882926 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner --allow-natives-syntax
```
-----------

Test:
`test/mjsunit/es6/classes-restricted-properties.js`
Command:
```bash
out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/mjsunit/es6/classes-restricted-properties.js --random-seed=-827882926 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner --allow-natives-syntax
```
-----------

Test:
`test/mjsunit/es6/destructuring.js`
Command:
```bash
out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/mjsunit/es6/destructuring.js --random-seed=-827882926 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner
```
-----------

Test:
`test/mjsunit/es6/generator-destructuring.js`
Command:
```bash
 out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/mjsunit/es6/generator-destructuring.js --random-seed=-827882926 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner
```
-----------

Test:
`test/mjsunit/es6/generators-iteration.js`
Command:
```bash
out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/mjsunit/es6/generators-iteration.js --random-seed=-827882926 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner --expose-gc
```
-----------

Test:
`test/mjsunit/es6/proxies-set.js`
Command:
```bash
out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/mjsunit/es6/proxies-set.js --random-seed=-827882926 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner
```
-----------

Test:
`test/mjsunit/es6/unicode-escapes-in-regexps.js `
Command:
```bash
out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/mjsunit/es6/unicode-escapes-in-regexps.js --random-seed=-827882926 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner
```
-----------

Test:
`test/mjsunit/harmony/async-generators-basic.js`
Command:
```bash
 out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/mjsunit/harmony/async-generators-basic.js --random-seed=-827882926 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner --allow-natives-syntax
```
-----------

Test:
`test/mjsunit/harmony/dataview-accessors.js`
Command:
```bash
out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/mjsunit/harmony/dataview-accessors.js --random-seed=-827882926 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner
```
-----------

Test:
`test/mjsunit/harmony/regexp-property-invalid.js`
Command:
```bash
out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/mjsunit/harmony/regexp-property-invalid.js --random-seed=-827882926 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner
```
-----------

Test:
`test/mjsunit/regress/regress-1365.js`
Command:
```bash
out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/mjsunit/regress/regress-1365.js --random-seed=-827882926 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner --allow-natives-syntax
```
-----------

Test:
`test/mjsunit/regress/regress-1708.js`
Command:
```bash
out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/mjsunit/regress/regress-1708.js --random-seed=-827882926 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner --expose-gc --noincremental-marking --max-semi-space-size=1
```
-----------

Test:
`test/mjsunit/regress/regress-319722-ArrayBuffer.js`
Command:
```bash
out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/mjsunit/regress/regress-319722-ArrayBuffer.js --random-seed=-827882926 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner --nostress-opt --allow-natives-syntax --mock-arraybuffer-allocator
```
-----------

Test:
` test/mjsunit/regress/regress-324028.js `
Command:
```bash
out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/mjsunit/regress/regress-324028.js --random-seed=-827882926 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner
```
-----------

Test:
`test/mjsunit/regress/regress-crbug-781583.js`
Command:
```bash
out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/mjsunit/regress/regress-crbug-781583.js --random-seed=-827882926 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner --allow-natives-syntax
```
-----------

Test:
`test/mjsunit/wasm/anyref-globals-interpreter.js`
Command:
```bash
out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/mjsunit/wasm/anyref-globals-interpreter.js --random-seed=-827882926 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner --experimental-wasm-anyref --expose-gc --wasm-interpret-all
```
-----------

Test:
`test/mjsunit/wasm/errors.js`
Command:
```bash
 out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/mjsunit/wasm/errors.js --random-seed=-827882926 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner --expose-wasm --allow-natives-syntax
```
-----------

Test:
`test/mjsunit/wasm/gc-stress.js `
Command:
```bash
out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/mjsunit/wasm/gc-stress.js --random-seed=-827882926 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner --expose-wasm --gc-interval=500 --stress-compaction
```
-----------

Test:
`test/mjsunit/wasm/instantiate-module-basic.js`
Command:
```bash
out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/mjsunit/wasm/instantiate-module-basic.js --random-seed=-827882926 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner --expose-wasm --allow-natives-syntax
```
-----------

Test:
`test/mjsunit/wasm/many-modules.js`
Command:
```bash
out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/mjsunit/wasm/many-modules.js --random-seed=-827882926 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner --wasm-far-jump-table --no-stress-opt
```

**Comment by luyahan** [comment]
pr #139  be merged into new branch: `reiscv64`

**Comment by penguinwu** [comment]
All fixed by #139
